File: api/src/main/java/net/impactdev/gts/api/data/ResourceManager.java
package net.impactdev.gts.api.data;

public interface ResourceManager<T> {

    String getName();

    /**
     * Represents the item ID that'll be used to reference the associated Entry type. This is purely
     * for the creation of the listing entry,
     *
     * @return The Minecraft Item ID that represents the item that should be used for entry creation
     */
    String getItemID();

    /**
     * Fetches the deserializer responsible for deserializing the resource into its desired typing.
     *
     * @return
     */
    Storable.Deserializer<T> getDeserializer();

}

File: api/src/main/kotlin/net/impactdev/gts/api/data/ResourceManager.kt
package net.impactdev.gts.api.data

interface ResourceManager<T> {
    val name: String?

    /**
     * Represents the item ID that'll be used to reference the associated Entry type. This is purely
     * for the creation of the listing entry,
     *
     * @return The Minecraft Item ID that represents the item that should be used for entry creation
     */
    val itemID: String?

    /**
     *
     *
     * @return
     */
    val deserializer: Storable.Deserializer<T>?
}